core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -146,7 +457,8 @@ query::QsRestrictor::Ptr newRestrictor(
 {
     // Extract the literals, bailing out if we see a non-literal
     bool isValid = true;
-    std::for_each(values.begin(), values.end(), validateLiteral(isValid));
+    std::for_each(values.begin(), values.end(),
+        [&](query::ValueExprPtr p) { isValid = isValid && p != nullptr && not p->copyAsLiteral().empty(); });
```

Comment:
Better to restrict it to:
```
   [&isValid](uery::ValueExprPtr p) { ...
```

Reply:
Thanks! I'm still getting the hang of lambdas...